tests/unittests/CoreGraphics.drawing/CGContextTransparencyLayerTests.cpp

```diff
@@ -0,0 +1,183 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//****************************************************************************** 
+
+#import "DrawingTest.h"
+
+static void _strokeTwoCirclesInContext(CGContextRef context, CGRect bounds) {
+    CGPoint center = _CGRectGetCenter(bounds);
+    CGRect centerEllipseRect = _CGRectCenteredOnPoint({ 150, 150 }, center);
+    CGFloat translations[]{ -60.f, +60.f };
+
+    for (float xSlide : translations) {
+        CGRect translatedRect = CGRectApplyAffineTransform(centerEllipseRect, CGAffineTransformMakeTranslation(xSlide, 0));
+        CGContextStrokeEllipseInRect(context, translatedRect);
+    }
+}
+
+DRAW_TEST_F(CGContext, TransparencyLayerIsShadowedAllAtOnce, WhiteBackgroundTest) {
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    CGContextSetShadow(context, CGSize{ 10.f, 10.f }, 1.0);
+
+    CGContextBeginTransparencyLayer(context, nullptr);
+
+    CGContextSetRGBFillColor(context, 1.0, 0.0, 0.0, 1.0);
+    CGContextSetRGBStrokeColor(context, 1.0, 0.0, 0.0, 1.0);
+    CGContextSetLineWidth(context, 5);
+
+    CGPoint center = _CGRectGetCenter(bounds);
+
+    CGContextMoveToPoint(context, center.x - 50, center.y);
+    CGContextAddLineToPoint(context, center.x + 50, center.y);
+    CGContextStrokePath(context);
+
+    CGContextMoveToPoint(context, center.x, center.y - 50);
+    CGContextAddLineToPoint(context, center.x, center.y + 50);
+    CGContextStrokePath(context);
+
+    CGContextEndTransparencyLayer(context);
+}
+
+DRAW_TEST_F(CGContext, TransparencyLayerWithInnerShadowCanStack, WhiteBackgroundTest) {
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    CGContextSetShadow(context, CGSize{ 10.f, 10.f }, 1.0);
+
+    CGContextBeginTransparencyLayer(context, nullptr);
+
+    CGContextSetShadow(context, CGSize{ -10.f, -10.f }, 1.0);
+
+    CGContextSetRGBStrokeColor(context, 1.0, 0.0, 0.0, 1.0);
+    CGContextSetLineWidth(context, 5);
+
+    _strokeTwoCirclesInContext(context, bounds);
+
+    CGContextEndTransparencyLayer(context);
+}
+
+DRAW_TEST_F(CGContext, TransparencyLayerIsAlphaCompositedAllAtOnce, WhiteBackgroundTest) {
```

Comment:
It is a statement rather than a question:

Transparency Layer is Alpha-Composited All at Once.

---
In reply to: [90587231](https://github.com/Microsoft/WinObjC/pull/1496#discussion_r90587231) [](ancestors = 90587231)